pkg/sources/docker/docker.go

```diff
@@ -132,6 +132,7 @@ func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk, _ .
 	_ = workers.Wait()
 	if scanErrs.Count() > 0 {
 		ctx.Logger().V(2).Info("scan errors", "errors", scanErrs.String())
+		return errors.New(scanErrs.String())
```

Comment:
Question: Are we certain this approach is what we want? Adopting this logic means any single error during the scan will lead to its failure. We've encountered this issue with other sources and found it detrimental to the user experience. However, the situation with the Docker source might be different, and perhaps we are willing to accept that trade-off

Reply:
As an alternative, implementing `SourceUnitEnumChunker` would provide a way to differentiate between fatal and non-fatal errors. It would be a little more work than adding this single line, but it doesn't look like it would be too much more, and it's work we want to do anyway.l